### PR TITLE
feat: Added Full python API support for MDO3 model.

### PR DESCRIPTION
**Pull request by v12ganesh**
## Proposed changes

Please include a summary of the changes and any links to related issues. Please also include relevant motivation and context.

Addresses #\< fill in issue number here >

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Functionality update (non-breaking change which updates or changes existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] CI/CD update (an update to the CI/CD workflows, scripts, and/or configurations)
- [ ] Documentation update (an update to enhance the user experience when reading through the docs)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have followed the guidelines in the [CONTRIBUTING](https://github.com/tektronix/tm_devices/blob/main/CONTRIBUTING.md) document
- [x] I have signed the CLA
- [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/tektronix/tm_devices/pulls) for the same update/change
- [ ] I have created (or updated) an [Issue](https://github.com/tektronix/tm_devices/issues) to track the status of this update/change and updated the link in this PR description (see above in the **Proposed changes** section) using the wording `Addresses #<issue_number>`
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Basic linting passes locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added necessary documentation (if appropriate)
- [x] I have updated the Changelog with a brief description of my changes

### CODE REVIEW

File: src/tm_devices/commands/gen_ujuvb_mdo/measurement.py [hunk 1]
```diff
@@ -2717,12 +2715,12 @@ def source1(self) -> MeasurementMeasItemSource1:
         Description:
             - For SOURce1: This command specifies the source for all single channel measurements.
               For delay or phase measurements, This command specifies the waveform to measure
-              'from'. This is equivalent to setting the 'From:' waveform in the 'Measure Delay' side
-              menu or the 'Measure Phase' side menu. SOUrce is equivalent to SOURCE1. For SOUrce2:
-              This command specifies the waveform to measure 'to' when taking a delay measurement or
-              phase measurement. This is equivalent to setting the 'To:' waveform in the 'Measure
-              Delay' side menu or the 'Measure Phase' side menu. Measurements are specified by <x>,
-              which ranges from 1 to 8.
+              'from'. This is equivalent  to setting the 'From:' waveform in the 'Measure Delay'
```
Comment (nfelt14) [inline src/tm_devices/commands/gen_ujuvb_mdo/measurement.py:2718]: Why the extra whitespace? This probably happened in more than one place.

File: src/tm_devices/commands/gen_1lh2st_msodpo/search.py [hunk 2]
```diff
@@ -1117,42 +1078,6 @@ def polarity(self) -> SearchSearchItemTriggerATimeoutPolarity:
         """
         return self._polarity
 
-    @property
-    def source(self) -> SearchSearchItemTriggerATimeoutSource:
-        """Return the ``SEARCH:SEARCH<x>:TRIGger:A:TIMEOut:SOUrce`` command.
```
Comment (nfelt14) [inline src/tm_devices/commands/gen_1lh2st_msodpo/search.py:1122]: What happened to this command?

**Review by nfelt14 — CHANGES_REQUESTED** [review]

File: src/tm_devices/commands/gen_1lh2st_msodpo/search.py (hunk 2 above)
Comment (v12ganesh) [reply src/tm_devices/commands/gen_1lh2st_msodpo/search.py:1122]: This command is available in MDO, MSO, LPDMSO and TekscopePC.

Comment (nfelt14) [reply src/tm_devices/commands/gen_1lh2st_msodpo/search.py:1122]: So why was it removed from here?

**Review by ArunKumarTek — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -52,23 +53,17 @@ However, please read through all changes to be aware of what may potentially imp
 - _**<span style="color:red">BREAKING CHANGE</span>**_: Changed the behavior of the `expect_esr()` method to expect an integer error code input and an optional tuple of error messages to compare against the actual error code and messages returned by the `_get_errors()` private method.
 - _**<span style="color:orange">minor breaking change</span>**_: Converted the `device_type` property into an abstract, cached property to force all children of the `Device` class to specify what type of device they are.
 - Updated the auto-generated command mixin classes to no longer use an `__init__()` method to enable the driver API documentation to render in a more usable way.
-- Switched from using standard `print()` calls to using the `logging` module for all logging in the `tm_devices` package.
```
Comment (nfelt14) [inline CHANGELOG.md:55]: Why was this removed?

File: CHANGELOG.md [hunk 4]
```diff
@@ -52,23 +53,17 @@ However, please read through all changes to be aware of what may potentially imp
...
-    - A configuration function provides the ability to set different logging levels for stdout and file logging.
-    - The config file and environment variable can also be used to control the logging functionality.
-    - The debug logging from the `pyvisa` package is also included in the log file by default.
 
 ### Removed
 
-- _**<span style="color:red">BREAKING CHANGE</span>**_: Removed previously deprecated `TekScopeSW` alias to the `TekScopePC` class.
```
Comment (nfelt14) [inline CHANGELOG.md:62]: Why was this removed?

File: CHANGELOG.md [hunk 5]
```diff
@@ -52,23 +53,17 @@ However, please read through all changes to be aware of what may potentially imp
...
 - _**<span style="color:red">BREAKING CHANGE</span>**_: Removed Internal AFG methods from the `TekScopePC` driver, since they wouldn't have worked due to its lack of an IAFG.
 - _**<span style="color:red">BREAKING CHANGE</span>**_: Removed previously deprecated `DEVICE_DRIVER_MODEL_MAPPING` constant.
 - _**<span style="color:red">BREAKING CHANGE</span>**_: Removed the `DEVICE_TYPE_CLASSES` constant and the `device_type_classes.py` module.
 - _**<span style="color:red">BREAKING CHANGE</span>**_: Removed many hacky implementations of `total_channels` and `all_channel_names_list` properties from drivers that don't need them anymore.
 - _**<span style="color:red">BREAKING CHANGE</span>**_: Removed the `verify_values()`, `raise_failure()`, and `raise_error()` methods from all device drivers.
     - These methods have been converted to helper functions and can be imported from the `tm_devices.helpers` subpackage now.
-- _**<span style="color:red">BREAKING CHANGE</span>**_: Removed the `print_with_timestamp()` function since this functionality is now handled by the `logging` module.
```
Comment (nfelt14) [inline CHANGELOG.md:70]: Why was this removed?

**Review by nfelt14 — APPROVED** [review]
